src/thunderbird_accounts/mail/views.py

```diff
@@ -62,6 +62,12 @@ def sign_up_submit(request: HttpRequest):
     )
     account.save_app_password('Mail Clients', request.POST['app_password'])
 
+    try:
+        Email.objects.get(address=email_address)
+        return raise_form_error(request, reverse('sign_up'),  _('Requested email is already taken'))
+    except Email.DoesNotExist:
+        pass
```

Comment:
I think I forgot to add a comment, but can you move this up with the other validation? So we're not creating anything if we don't have a unique email.